File: 1823-find-the-winner-of-the-circular-game/1823-find-the-winner-of-the-circular-game.java
class Solution {
    public int findTheWinner(int n, int k) {
        Queue<Integer> que = new LinkedList<Integer>();
        for(int i = 0 ; i < n ; i++){
            que.add(i);
        }
        while(que.size() > 1){
            for(int i = 1 ; i < k ; i++){
                que.add(que.remove());
            }
            que.remove();
            
        }
        return (int)que.peek() + 1;
    }
}

File: 1823-find-the-winner-of-the-circular-game/1823-find-the-winner-of-the-circular-game.cpp
class Solution {
public:
    int findTheWinner(int n, int k) {
        queue<int> que;
        for(int i = 1 ; i <=n ; i ++){
            que.push(i);
        }
        while(que.size() > 1){
            for(int i = 1 ; i < k ; i++){
                int x = que.front();
                que.pop();
                que.push(x);
            }
            que.pop();
        }
        return que.front();
    }
};